### PR TITLE
Remove signed HTTP request support

### PR DESCRIPTION
**Pull request by achamayou**
This completes #3875 by removing signed HTTP request support from CCF, after its deprecation in 3.0.

- [x] remove tests that make use of http request signing, see comment for explanation
- [x] resolve problem in membership and governance test with signed GETs (see below for discussion)
- [x] additional pass to remove dead code

### CODE REVIEW

File: src/node/rpc/member_frontend.h [hunk 1]
```diff
@@ -447,9 +447,9 @@ namespace ccf
         cose_recent_proposals->put(key, proposal_id);
         // Only keep the most recent window_size proposals, to avoid
         // unbounded memory usage
-        if (replay_keys.size() >= window_size)
+        if (replay_keys.size() >= (window_size - 1) /* We just added one */)
```
Comment (achamayou) [inline src/node/rpc/member_frontend.h:450]: This was previously off by one, hidden by the fact that the test was the first to submit COSE proposals.

File: tests/governance_js.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def check_msg_type(verb, path, name):
             ("POST", "/gov/ack", "ack"),
             ("POST", "/gov/ack/update_state_digest", "state_digest"),
             ("POST", "/gov/recovery_share", "recovery_share"),
-            ("GET", "/gov/recovery_share", "encrypted_recovery_share"),
+            # ("GET", "/gov/recovery_share", "encrypted_recovery_share"),
```
[the remark this answers is not in the document]
Comment (achamayou) [reply tests/governance_js.py:102]: Went with option 3 in #5145

**Comment by achamayou** [comment]
t_cose does not support passing in custom header parameters [1][2] when signing (nor when verifying, but we can and do re-parse there with QCBOR), which we need for CCF governance.

[1] https://github.com/laurencelundblade/t_cose/blob/master/examples/t_cose_basic_example_ossl.c#L262
[2] https://github.com/laurencelundblade/t_cose/issues/5

**Comment by achamayou** [comment]
The other problem with the proposal test is that it relies on being able to submit concurrent identical proposals, which is now caught by the replay proposal protection.

**Comment by achamayou** [comment]
Following discussion with @jumaffre, I am removing the proposal_id_test, because:

1. Updating create_signed_request to use COSE Sign1 requires substantial changes to t_cose that we don't have time to make now.
2. Even with this change, the test would not work with the new replay protection functionality (tested separately).
3. The logic covered by this test uniquely is relatively trivial (History rollback), generic KV conflict functionality is tested separately.

File: doc/governance/hsm_keys.rst [hunk 3]
```diff
@@ -117,45 +117,6 @@ Like ``ccf_cose_sign1``, the output can be sent directly to the service via curl
         "state": "Open"
     }
 
-HTTP Signing
```
Comment (jumaffre) [inline doc/governance/hsm_keys.rst:120]: I think we may want to keep this section with a warning that this scheme was removed in 4.x.

File: doc/governance/proposals.rst [hunk 4]
```diff
@@ -278,18 +278,6 @@ For example, ``member1`` may submit a proposal to add a new member (``member4``)
       "state": "Open"
     }
 
-Or alternatively, with the old signature method:
```
Comment (jumaffre) [inline doc/governance/proposals.rst:281]: Same here I think, we should add a warning at the top of the page to say that this has been removed in 4.x.

File: doc/schemas/app_openapi.json [hunk 5]
```diff
@@ -286,16 +286,6 @@
         "description": "Request payload must be a COSE Sign1 document, with expected protected headers. Signer must be a member identity registered with this service.",
         "scheme": "cose_sign1",
         "type": "http"
-      },
-      "member_signature": {
-        "description": "Request must be signed according to the HTTP Signature scheme. The signer must be a member identity registered with this service.",
-        "scheme": "signature",
-        "type": "http"
-      },
-      "user_signature": {
```
Comment (jumaffre) [inline doc/schemas/app_openapi.json:295]: Do we have an equivalent COSE signature scheme that users can use for apps, in both JS and C++?

File: doc/use_apps/issue_commands.rst [hunk 6]
```diff
@@ -30,24 +30,7 @@ The response body (the JSON value ``true``) indicates that the request was execu
 Signing
 -------
 
-In some situations CCF requires signed requests, for example for member votes. Two signing schemes are supported as of 3.x.
-
-HTTP Signatures
-~~~~~~~~~~~~~~~
-
-An implementation of `IETF HTTP Signatures draft RFC <https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-08>`_ , but
```
Comment (jumaffre) [inline doc/use_apps/issue_commands.rst:38]: Forget my other commends; this is probably the place to have the warning!

File: src/node/rpc/member_frontend.h [hunk 7]
```diff
@@ -91,24 +90,6 @@ namespace ccf
   class MemberEndpoints : public CommonEndpointRegistry
   {
   private:
-    // Wrapper for reporting errors, which both logs them under the [gov] tag
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:94]: I don't think this should go?

File: src/node/rpc/member_frontend.h [hunk 8]
```diff
@@ -505,7 +486,7 @@ namespace ccf
 
       auto getter =
         [&, table](endpoints::ReadOnlyEndpointContext& ctx, nlohmann::json&&) {
-          GOV_TRACE_FMT("Called getter for {}", table.get_name());
+          LOG_TRACE_FMT("Called getter for {}", table.get_name());
```
Comment (jumaffre) [inline src/node/rpc/member_frontend.h:489]: Merge issue?

File: doc/schemas/app_openapi.json (hunk 5 above)
Comment (achamayou) [reply doc/schemas/app_openapi.json:295]: Not at the moment, no.

File: src/node/rpc/member_frontend.h (hunk 7 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h]: Yes, sorry, bad merge. Fixed now.

File: src/node/rpc/member_frontend.h (hunk 8 above)
Comment (achamayou) [reply src/node/rpc/member_frontend.h]: Yes, sorry, bad merge. Fixed now.

**Review by jumaffre — COMMENTED** [review]
This needs a changelog entry and I think some warning in the documentation so that this feature doesn't disappear silently (even though we've warned users that this was deprecated!). It's also not clear if users can make use of the new COSE signatures via built-in authentication policies, in both C++ and JS?

**Review by jumaffre — APPROVED** [review]

**Pull request merged**